interpreter/src/lib.rs

```diff
@@ -37,44 +39,117 @@ impl State {
         self.memory.insert(x.to_string(), value);
     }
 
-    pub fn write(&mut self, x: &str, value: u32) {
+    pub fn write(&mut self, x: &str, value: u32, thread: &str) {
+
         match self.memory_model {
             MemoryModel::Sc => {
                 self.memory.insert(x.to_string(), value);
             }
             MemoryModel::Tso => {
-                todo!()
+                if let Some(buffer) = self.write_buffers.get_mut(thread) {
+                    buffer.insert(x.to_string(), value);
+                } else {
+                    self.memory.insert(x.to_string(), value);
+                }
```

Comment:
This should always insert the value into the write buffer and never directly into memory